### PR TITLE
Fix RerunLastFailures problem. Appear the perfix only one time 

### PR DESCRIPTION
**Pull request by Eqerem-Hena**
Fix the #1342 issue.

### CODE REVIEW

**Comment by fhoeben** [comment]
@Eqerem-Hena Thanks a lot for your PR!

I believe you are generating unique names for each re-rerun by adding a counter at the end, correct?
To me this seems overkill as it keeps making a lot of pages if the rerun is retried often.

Like I stated in the issue just having one rerun per page/suite seems fine. So this PR could be much simpler: just don't add the `RerunLastFailures_` prefix if the  full page name already starts with that.
Do you think that is too limited?

**Comment by fhoeben** [comment]
It also seems like some of the existing tests failed, can that be related to your change or did we let something else slip through and break that already?

**Comment by Eqerem-Hena** [comment]
> @Eqerem-Hena Thanks a lot for your PR!
> 
> I believe you are generating unique names for each re-rerun by adding a counter at the end, correct? To me this seems overkill as it keeps making a lot of pages if the rerun is retried often.
> 
> Like I stated in the issue just having one rerun per page/suite seems fine. So this PR could be much simpler: just don't add the `RerunLastFailures_` prefix if the full page name already starts with that. Do you think that is too limited?

I think it is a better idea to add the counter at the end, as in this way it will be more understandable to the user which rerun page is.

**Comment by Eqerem-Hena** [comment]
> It also seems like some of the existing tests failed, can that be related to your change or did we let something else slip through and break that already?

I don't think so. It's probably from a previous commit.

**Comment by fhoeben** [comment]
The original submitter of the issue indicated:
> This would help keep my FitNesseRoot directory from exploding...

So they prefer not to have an additional file for each rerun.

**Comment by Eqerem-Hena** [comment]
> The original submitter of the issue indicated:
> 
> > This would help keep my FitNesseRoot directory from exploding...
> 
> So they prefer not to have an additional file for each rerun.

I did a new commit. Please, check it out!

File: src/fitnesse/responders/run/SuiteResponder.java [hunk 1]
```diff
@@ -295,7 +295,12 @@ protected String getRerunPageName() {
     PageCrawler pageCrawler = page.getPageCrawler();
     WikiPagePath fullPath = pageCrawler.getFullPath();
     String fullPathName = PathParser.render(fullPath);
-    return "RerunLastFailures_"+fullPathName.replace(".","-");
+    if (fullPathName.startsWith("RerunLastFailures_")) {
+      String newFullPathName = fullPathName.replace(".", "-");
```
Comment (fhoeben) [inline src/fitnesse/responders/run/SuiteResponder.java:299]: I think it makes more sense to to the replace on line 297 to remove the duplicated code from both branches

File: test/fitnesse/responders/run/SuiteResponderTest.java [hunk 2]
```diff
@@ -564,7 +570,29 @@ public void loadsCustomFormatters() throws Exception {
...
+  
+  @Test
+  public void testGetRerunPageName_withRerunPrefix() {
+
+    String fullPathName = "RerunLastFailures_SomePage";
+    when(PathParser.render(wikiPagePath)).thenReturn(fullPathName);
+
+    String result = responder.getRerunPageName();
+
+    assertEquals("RerunLastFailures_SomePage".replace(".", "-"), result);
```
Comment (fhoeben) [inline test/fitnesse/responders/run/SuiteResponderTest.java:582]:
```suggestion
    assertEquals("RerunLastFailures_SomePage", result);
```

File: test/fitnesse/responders/run/SuiteResponderTest.java [hunk 3]
```diff
@@ -564,7 +570,29 @@ public void loadsCustomFormatters() throws Exception {
...
 
+  @Test
+  public void testGetRerunPageName_withoutRerunPrefix() {
+
+    String fullPathName = "SomeOtherPage";
+    when(PathParser.render(wikiPagePath)).thenReturn(fullPathName);
+
+    String result = responder.getRerunPageName();
+
+    assertEquals("RerunLastFailures_SomeOtherPage".replace(".", "-"), result);
```
Comment (fhoeben) [inline test/fitnesse/responders/run/SuiteResponderTest.java:593]:
```suggestion
    assertEquals("RerunLastFailures_SomeOtherPage", result);
```

File: test/fitnesse/responders/run/SuiteResponderTest.java [hunk 4]
```diff
@@ -564,7 +570,29 @@ public void loadsCustomFormatters() throws Exception {
 
     assertTrue(FooFormatter.initialized);
   }
+
```
Comment (fhoeben) [inline test/fitnesse/responders/run/SuiteResponderTest.java:573]: It is a good idea to add the replacement of `.` by `-` in that page names, but I would like to see that in separate test cases where the `fullPathName` actually contains a period and then also just do an `assertEquals` where the expect value is a literal with the periods replaced.

In the current setup the `replace` in the assertion doesn't do anything and the test doesn't make explicit what it is expecting

**Review by fhoeben — CHANGES_REQUESTED** [review]
Looks really good. Thank you. Some minor comments and then we can merge

**Comment by fhoeben** [comment]
Could you also update the release notes?

**Comment by fhoeben** [comment]
Cleaned up the tests in #1519

**Pull request closed without merging**